modelling/feat_engineer/feature_engineering.py

```diff
@@ -10,10 +11,54 @@
 from sklearn.feature_selection import SelectKBest, f_classif
 import pywt
 from ..lamonaca_and_nemcova.utils import spo2_estimation, get_peak_height_slope
+from tqdm import tqdm
+
+
+def _eval_feat_engineering(all_features):
+    # Little helper function I wrote for some analysis comparing HR and SpO2 as a target
+
+    # https://github.com/gianlucatruda/Spo2_evaluation/issues/4
+    from sklearn.metrics import mean_squared_error as mse
+    from sklearn.linear_model import LassoLars, LinearRegression
+    from sklearn.neural_network import MLPRegressor
+    from xgboost import XGBRegressor
+    from tqdm import tqdm
+    from sklearn.model_selection import KFold
+    import json
+    from pandas.io.json import json_normalize
+
+    import warnings
+    warnings.filterwarnings(action='ignore', category=UserWarning)
+    warnings.filterwarnings(action='ignore', category=RuntimeWarning)
+
+    targets = ['spo2', 'hr']
+    models = [XGBRegressor, LassoLars, LinearRegression]
+    model_names = ['xgboost', 'lasso', 'lr']
+    K_vals = [2, 3, 5, 10, 15, 20, 30, 40, 50, 60, 70, 80]
+
+    results = {'target': [], 'model': [], 'k': [], 'mse': []}
+
+    for target in targets:
+        print(f"Evaluating {target}...")
+        for K in tqdm(K_vals, total=2*len(K_vals)):
+            _X = all_features.drop(targets, axis=1).values
+            _y = all_features[target].values
+            kbest = SelectKBest(f_classif, k=K).fit(_X, _y)
```

Comment:
Would be a good idea to pass the best features further for model training (in case we want to do it in a pipeline).

Reply:
Yeah, definitely. This function was just for me to evaluate HR vs SpO2 (which is why the name is underscored). But I'll probably build our evaluation pipeline from this code, so I left it in for now.